apps/authentication-portal/src/main/webapp/basicauth.jsp

```diff
@@ -391,23 +395,41 @@
     </div>
     <% } %>
     <%!
-        private String getRecoverAccountUrl (
-            String identityMgtEndpointContext,
-            String urlEncodedURL,
-            boolean isUsernameRecovery,
-            String urlParameters) {
-
-            return identityMgtEndpointContext + "/recoveraccountrouter.do?" + urlParameters +
-                "&isUsernameRecovery=" + isUsernameRecovery + "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+        private String getRecoverAccountUrl(String identityMgtEndpointContext, String urlEncodedURL,
+                boolean isUsernameRecovery, String urlParameters) {
+
+            return buildUrl(identityMgtEndpointContext) + ACCOUNT_RECOVERY_ENDPOINT_RECOVER + "?" + urlParameters
+                    + "&isUsernameRecovery=" + isUsernameRecovery + "&callback=" + Encode
+                    .forHtmlAttribute(urlEncodedURL);
+        }
+
+        private String getRegistrationUrl(String identityMgtEndpointContext, String urlEncodedURL,
+                String urlParameters) {
+
+            return buildUrl(identityMgtEndpointContext) + ACCOUNT_RECOVERY_ENDPOINT_REGISTER + "?" + urlParameters
+                    + "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
         }
 
-        private String getRegistrationUrl (
-            String identityMgtEndpointContext,
-            String urlEncodedURL,
-            String urlParameters) {
+        private String buildUrl(String identityMgtEndpointContext) {
 
-            return identityMgtEndpointContext + "/register.do?" + urlParameters +
-                "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+            String url;
+            if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+                url = buildTenantQualifiedUrl(ACCOUNT_RECOVERY_ENDPOINT);
+            } else if (StringUtils.isNotBlank(identityMgtEndpointContext)) {
+                url = identityMgtEndpointContext;
+            } else {
+                url = getServerURL(ACCOUNT_RECOVERY_ENDPOINT, true, true);
```

Comment:
Rather than calling legacy API we should call the new API right

Reply:
Looks to me we should change the if else ordering

Reply:
Updated the logic as discussed.